packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -0,0 +1,283 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import { getScope, getSourceCode } from '@typescript-eslint/utils/eslint-utils';
+import type { RuleFix, SourceCode } from '@typescript-eslint/utils/ts-eslint';
+import * as tsutils from 'ts-api-utils';
+import type { Type } from 'typescript';
+
+import {
+  createRule,
+  getConstrainedTypeAtLocation,
+  getParserServices,
+  getStaticValue,
+  nullThrows,
+} from '../util';
+
+export default createRule({
+  name: 'prefer-find',
+  meta: {
+    docs: {
+      description:
+        'Enforce the use of Array.prototype.find() over Array.prototype.filter() followed by [0] when looking for a single result',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      preferFind: 'Use .find(...) instead of .filter(...)[0]',
+      preferFindSuggestion: 'Use .find(...) instead of .filter(...)[0]',
+    },
+    schema: [],
+    type: 'suggestion',
+    hasSuggestions: true,
+  },
+
+  defaultOptions: [],
+
+  create(context) {
+    const globalScope = getScope(context);
+    const services = getParserServices(context);
+    const checker = services.program.getTypeChecker();
+
+    function parseIfArrayFilterExpression(
+      expression: TSESTree.Expression,
+    ):
+      | { isBracketSyntaxForFilter: boolean; filterNode: TSESTree.Node }
```

Comment:
[Refactor] I just filed https://github.com/typescript-eslint/typescript-eslint/issues/8320 thinking on inline object literals. WDYT?

Reply:
yep. You make a fair point. changed!